cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,147 @@
+        self.assertEqual(res.data, serializer.data)
+
+    def test_movie_by_actors(self):
+        actor1 = sample_actor(first_name="Tom", last_name="Holland")
+        actor2 = sample_actor(first_name="Ryan", last_name="Reynolds")
+
+        movie_with_actor1 = sample_movie()
+        movie_with_actor2 = sample_movie()
+        movie_without_actor = sample_movie()
+        movie_with_actor1.actors.add(actor1)
+        movie_with_actor2.actors.add(actor2)
+
+        res = self.client.get(MOVIE_URL, {"actors": f"{actor1.id}, {actor2.id}"})
+
+        serializer_with_actor1 = MovieListSerializer(movie_with_actor1)
+        serializer_with_actor2 = MovieListSerializer(movie_with_actor2)
+        serializer_without_actor = MovieListSerializer(movie_without_actor)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(res.data), 2)
+
+        self.assertIn(serializer_with_actor1.data, res.data)
+        self.assertIn(serializer_with_actor2.data, res.data)
+        self.assertNotIn(serializer_without_actor.data, res.data)
+
+    def test_movie_by_genres(self):
+        genre1 = sample_genre(name="Horror")
+        genre2 = sample_genre(name="Fantasy")
+
+        movie_with_genre1 = sample_movie()
+        movie_with_genre2 = sample_movie()
+        movie_without_genre = sample_movie()
+        movie_with_genre1.genres.add(genre1)
+        movie_with_genre2.genres.add(genre2)
+
+        res = self.client.get(MOVIE_URL, {"genres": f"{genre1.id}, {genre2.id}"})
+
+        serializer_with_genre1 = MovieListSerializer(movie_with_genre1)
+        serializer_with_genre2 = MovieListSerializer(movie_with_genre2)
+        serializer_without_genre = MovieListSerializer(movie_without_genre)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(res.data), 2)
+
+        self.assertIn(serializer_with_genre1.data, res.data)
+        self.assertIn(serializer_with_genre2.data, res.data)
+        self.assertNotIn(serializer_without_genre.data, res.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        movie.genres.add(sample_genre())
+        movie.actors.add(sample_actor())
+        res = self.client.get(detail_url(movie.id))
+        serializer = MovieDetailSerializer(movie)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_post_retrieve_movie(self):
+        payload = {
+            "title": "Test movie",
+            "description": "Test description",
+            "duration": 110,
+        }
+        res = self.client.post(MOVIE_URL, payload)
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_delete_retrieve_movie(self):
+        movie = sample_movie()
+        res = self.client.delete(detail_url(movie.id))
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
```

Comment:
The `test_delete_retrieve_movie` method expects a `403 FORBIDDEN` status code. Verify that this is the correct response for unauthorized DELETE requests. If the API should allow authenticated users to delete movies, consider modifying the test or the API permissions.